src/Planner/PlannerActionsVisitor.cpp

```diff
@@ -500,7 +501,39 @@ class PlannerActionsVisitorImpl
     ActionsDAG::NodeRawConstPtrs visit(QueryTreeNodePtr expression_node);
 
 private:
-    using NodeNameAndNodeMinLevel = std::pair<std::string, size_t>;
+
+    class Levels
+    {
+    public:
+        explicit Levels(size_t level) { set(level); }
+
+        void set(size_t level)
+        {
+            check(level);
+            mask |= uint64_t(1) << level;
+        }
+
+        void reset(size_t level)
+        {
+            check(level);
+            mask &= ~(uint64_t(1) << level);
+        }
+
+        void add(Levels levels) { mask |= levels.mask; }
+
+        size_t max() const { return 63 - __builtin_clzll(mask); }
+
+    private:
+        uint64_t mask = 0;
+
+        void check(size_t level)
+        {
+            if (level >= 63)
+                throw Exception(ErrorCodes::INCORRECT_QUERY, "Maximum lambda depth exceeded. Maximum 63.");
+        }
```

Comment:
Shouldn't this be `level > 63`? I think 63 should be possible (`1 << 63` is valid), and as far as I understood the exception message also suggests that (maximum 63, maximum is exceeded => bigger than 63).